test/common_test/collisions.cc

```diff
@@ -141,6 +146,106 @@ TYPED_TEST(CollisionTest, MeshAndPlane)
   }
 }
 
+struct CollisionMeshFeaturesList : gz::physics::FeatureList<
+  gz::physics::sdf::ConstructSdfModel,
+  gz::physics::sdf::ConstructSdfWorld,
+  gz::physics::LinkFrameSemantics,
+  gz::physics::ForwardStep,
+  gz::physics::GetEntities
+> { };
+
+template <class T>
+class CollisionMeshTest :
+  public CollisionTest<T>{};
+using CollisionMeshTestTypes =
+  ::testing::Types<CollisionMeshFeaturesList>;
+TYPED_TEST_SUITE(CollisionMeshTest,
+                 CollisionMeshTestTypes);
+
+TYPED_TEST(CollisionMeshTest, MeshDecomposition)
+{
+  // Load an optimized mesh, drop it from some height,
+  // and verify it collides with the ground plane
+  std::string modelOptimizedStr = R"(
+  <sdf version="1.11">
+    <model name="model_optimized">
+      <pose>0 0 2.0 0 0 0</pose>
+      <link name="body">
+        <collision name="collision">
+          <geometry>
+            <mesh optimization="convex_decomposition">
```

Comment:
this test could be generalized to include other values of `//mesh/@optimization` (both `convex_hull` and the default `""` case)

Reply:
generalized test to include other optimization methods. [1f2aaa7](https://github.com/gazebosim/gz-physics/pull/606/commits/1f2aaa7a24756c51f56b5d03ad56958926d38630)